src/linux/Payload.Linux/build.sh

```diff
@@ -32,17 +38,106 @@ case "$i" in
 esac
 done
 
+# Directories
+THISDIR="$( cd "$(dirname "$0")" ; pwd -P )"
+ROOT="$( cd "$THISDIR"/../../.. ; pwd -P )"
+SRC="$ROOT/src"
+OUT="$ROOT/out"
+GCM_SRC="$SRC/shared/Git-Credential-Manager"
+# PAYLOAD_SRC="$SRC/linux/Payload.Linux"
+PAYLOAD_OUT="$OUT/linux/"
+
+# Build parameters
+FRAMEWORK=netcoreapp3.1
+RUNTIME=linux-x64
+
 # Perform pre-execution checks
 CONFIGURATION="${CONFIGURATION:=Debug}"
 if [ -z "$VERSION" ]; then
     die "--version was not set"
 fi
 
-PAYLOAD="$PAYLOAD_OUT/tar/$CONFIGURATION/payload"
-TAROUT="$PAYLOAD_OUT/tar/$CONFIGURATION/gcmcore-linux-x86_64-$VERSION.tar.gz"
+ARCH="`dpkg-architecture -q DEB_HOST_ARCH`"
+if test -z "$ARCH"; then
+  die "Could not determine host architecture!"
+fi
+
+PAYLOAD="$PAYLOAD_OUT/payload/$CONFIGURATION"
+TAROUT="$PAYLOAD_OUT/gcmcore-linux_$ARCH.$CONFIGURATION.$VERSION.tar.gz"
+DEBPKG="$PAYLOAD_OUT/gcmcore-linux/"
+DEBOUT="$PAYLOAD_OUT/gcmcore-linux_$ARCH.$CONFIGURATION.$VERSION.deb"
+SYMBOLOUT="$PAYLOAD.sym"
+
+# Cleanup any old payload directory
+if [ -d "$PAYLOAD" ]; then
+    echo "Cleaning old payload directory '$PAYLOAD'..."
+    rm -rf "$PAYLOAD"
+fi
+
+# Ensure directories exists
+mkdir -p "$PAYLOAD" "$SYMBOLOUT" "$DEBPKG"
+
+# Publish core application executables
+echo "Publishing core application..."
+dotnet publish "$GCM_SRC" \
+	--configuration="$CONFIGURATION" \
+	--framework="$FRAMEWORK" \
+	--runtime="$RUNTIME" \
+    --self-contained=true \
+    "/p:PublishSingleFile=True" \
+	--output="$(make_absolute "$PAYLOAD")" || exit 1
+
+# Collect symbols
+echo "Collecting managed symbols..."
+mv "$PAYLOAD"/*.pdb "$SYMBOLOUT" || exit 1
+
+echo "Build complete."
+
+#####################################################################
+# PACKING
+#####################################################################
+echo "Packing Payload.Linux..."
+# Cleanup any old archive files
+if [ -e "$TAROUT" ]; then
+    echo "Deleteing old archive '$TAROUT'..."
+    rm "$TAROUT"
+fi
+
+# Ensure the parent directory for the archive exists
+mkdir -p "$(dirname "$TAROUT")"
+
+# Set full read, write, execute permissions for owner and just read and execute permissions for group and other
+echo "Setting file permissions..."
+/bin/chmod -R 755 "$PAYLOAD" || exit 1
+
+# Build tarball
+echo "Building archive..."
+pushd "$PAYLOAD"
+tar -czvf "$TAROUT" * || exit 1
+popd
+
+# Build .deb
+INSTALL_TO="$DEBPKG/usr/bin/"
+mkdir -p "$DEBPKG/DEBIAN" "$INSTALL_TO"
+
+# make the debian control file
+cat >"$DEBPKG/DEBIAN/control" <<EOF
+Package: gcmcore
```

Comment:
Question: do we want the package to be `gcmcore` or `gcm-core` or `git-credential-manager-core`?

Reply:
@mjcheetham I'm guessing we should follow the precedent for the windows and mac distro naming? Which I think is `git-credential-manager-core` ?

Reply:
Are alias supported? I realise `git-credential-manager-core` is a mouthful, but it is the official name (and I'm a stickler for formalities 😛)..